test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -60,5 +63,92 @@ public void AutoFitRegressionTest()
 
             Assert.True(result.RunDetails.Max(i => i.ValidationMetrics.RSquared > 0.9));
         }
+
+        [Fact]
+        public void AutoFitRecommendationTest()
+        {
+            // Specific column names of the considered data set
+            string labelColumnName = "Label";
+            string userColumnName = "User";
+            string itemColumnName = "Item";
+            string scoreColumnName = "Score";
+            MLContext mlContext = new MLContext();
+
+            // STEP 1: Load data
+            var reader = new TextLoader(mlContext, GetLoaderArgs(labelColumnName, userColumnName, itemColumnName));
+            var trainDataView = reader.Load(new MultiFileSource(GetDataPath(TestDatasets.trivialMatrixFactorization.trainFilename)));
+            var testDataView = reader.Load(new MultiFileSource(GetDataPath(TestDatasets.trivialMatrixFactorization.testFilename)));
+
+            // STEP 2: Run AutoML experiment
+            ExperimentResult<RegressionMetrics> experimentResult = mlContext.Auto()
+                .CreateRecommendationExperiment(5)
+                .Execute(trainDataView, testDataView,
+                    new ColumnInformation() { 
+                        LabelColumnName = labelColumnName,
+                        MatrixColumnIndexColumnName = userColumnName,
+                        MatrixRowIndexColumnName = itemColumnName
+                    });
+
+            RunDetail<RegressionMetrics> bestRun = experimentResult.BestRun;
+            Assert.True(experimentResult.RunDetails.Count() > 1);
+            Assert.NotNull(bestRun.ValidationMetrics);
+            Assert.True(experimentResult.RunDetails.Max(i => i.ValidationMetrics.RSquared != 0));
+
+            var outputSchema = bestRun.Model.GetOutputSchema(trainDataView.Schema);
+            var expectedOutputNames = new string[] { labelColumnName, userColumnName, userColumnName, itemColumnName, itemColumnName, scoreColumnName };
```

Comment:
- [ ] revisit MatrixColumn.Apply method which applies ValueToKey mapping

Given test data with three columns: 
```
Label	Row	Column
...
1	14	35
1	14	36
1	14	37
1	14	38
1	14	39
3	15	0
3	15	1
3	15	3
3	15	4
3	15	5
3	15	6
...
```

As shown in line 98, we will get output schema with 5 columns. That is due to line 204 in the diffs of TransformInference.cs

I'm not sure if this is OK.
It looks odd compared to test MatrixFactorizationSimpleTrainAndPredict in MatrixFactorizationTests where
```
            var expectedOutputNames = new string[] { labelColumnName, userColumnName, itemColumnName, scoreColumnName };
```